File: static/code/sample/tethered/gdl-soradiofm-a.cs
// In this sample:
// Allow to set volume and channel on RadioFM module

using GHIElectronics.DUELink;

var availablePort = DUELinkController.GetConnectionPort();
var duelink = new DUELinkController(availablePort);

void SetVolume(int volume) {
    if (volume >= 0 && volume <= 100) {
        duelink.Engine.ExecuteCommand($"SetVol({volume})");
    }
}
void SetChannel(float channel) {
    duelink.Engine.ExecuteCommand($"SetChan({channel})");
}

SetVolume(50); // set 50%
SetChannel(100.3f); // set channel 100.3 

File: static/code/sample/tethered/gdl-soradiofm-a.py
# In this sample:
# Allows setting volume and channel on the RadioFM module

import time
from DUELink.DUELinkController import DUELinkController

availablePort = DUELinkController.GetConnectionPort()
duelink = DUELinkController(availablePort)

# method
def SetVolume(volume: int):
    """Set the volume on the RadioFM module (0-100)."""
    if 0 <= volume <= 100:
        duelink.Engine.ExecuteCommand(f"SetVol({volume})")

def SetChannel(channel: float):
    """Set the channel on the RadioFM module."""
    duelink.Engine.ExecuteCommand(f"SetChan({channel})")

# Example usage
SetVolume(50)    # Set volume to 50%
SetChannel(100.3)  # Set channel to 100.3